### DISCUSSION TITLE
GAT for biparite graph vs BipariteGraphOperator

### DISCUSSION THREAD

**Discussion by lisiq** [pyg-team/pytorch_geometric#4415] [Q&A] [answered]
Hi everyone, I am a newbie in PyTorch Geometric. Thank you for the amazing job that you have done!

I found an old discussion [here](https://github.com/pyg-team/pytorch_geometric/issues/118) about bipartite graph support.
I also saw that GAT works for bipartite graphs. I noticed a difference between the output dimensions of these two layers. 

Suppose that we have a directed bipartite graph with the first independent set having 100 nodes pointing to 10 nodes in the other independent set. If num_classes is 3 then GAT returns an output of shape (100,3) and the BipartiteGraphOperator from the link above return an output of dimension of shape (10, 3).

My question is the following. Is it possible to make GAT have the same output dimension as the BipartiteGraphOperator layer? If not is there any other method that does this? 

Thank you very much!

**Comment by lisiq** [comment, marked as answer]
Everything is clear now, it was working as it should but I had a missing piece that was causing all of this. Thank you!

**Reply by rusty1s** [reply]
Sorry for my delayed response. Glad that you resolved it on your own. In general, we can only ensure that the output feature dimension matches while the node sizes of source and destination nodes stay as they are.

**Reply by lisiq** [reply]
Thank you for your reply! I had an additional question. Is there any implementation that works on multipartite graphs?
Would only adding x=(x_1, x_2, x_3) and size=(N1, N2, N3) in [this](https://github.com/pyg-team/pytorch_geometric/issues/118) implementation make sense/work?
